### PR TITLE
chore(expo): update expo version to 49

### PR DESCRIPTION
**Pull request by d4v3d4vE**
- ran `yarn add expo@^49.0.0`
- ran `npx expo install --fix` to upgrade all dependencies to match SDK 49
- added `"expo-localization"` to `app.json` as part of the expo 49 update
- https://blog.expo.dev/expo-sdk-49-c6d398cdf740

- updated `react-navigation` from 5.X to 6.X manually as part of the expo 49 update -> changed the version numbers of `react-navigation` from 5.X to 6.X inside the `package.json`, deleted the `node_modules` folder, deleted the `yarn.lock` file and ran the `yarn` command to ensure a clean install
- adjusted the headerStyle in `screenOptions.tsx` to match the `react-navigation` 6.X to the before used design with `react-navigation` 5.X
- changed in `MainTabNavigator.tsx` the `tabBarOptions` to `screenOptions` and the options names to match with the `react-navigation` 6.X update
- added `headerShown: false` to `screenOptions` in `MainTabNavigator.tsx` to ensure the before used design with `react-navigation` 5.X and not show a tab screen header with the new `react-navigation` 6.X update
- added `screenOptions` to the `Drawer.Navigator` in `DrawerNavigator.tsx` with `drawerPosition: "right"` and `headerShown: false` to ensure the before used design with `react-navigation` 5.X and not show a drawer screen header with the new `react-navigation` 6.X update

- changed the navigation from drawer to tabnavigation

- to setup the new expo version execute the following commands in the project folder: 

1. run `yarn prebuild`

2. run `npx pod-install`

3. run `yarn ios -d`

SVA-1081

| TabNav before | TabNav now | Drawer before | Drawer now |
| ------------- | ------------- | ------------- | ------------- |
| Simulator Screenshot - iPhone 14 Pro - 2023-08-18 at 18 12 36 | Simulator Screenshot - iPhone 14 Pro - 2023-08-18 at 18 14 03 | Simulator Screenshot - iPhone 14 Pro - 2023-08-18 at 18 10 30 | Simulator Screenshot - iPhone 14 Pro - 2023-08-18 at 18 13 03 |
|  |  | Simulator Screenshot - iPhone 14 Pro - 2023-08-18 at 18 10 34 | Simulator Screenshot - iPhone 14 Pro - 2023-08-18 at 18 13 07 |

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -37,15 +37,15 @@
...
-    "@react-native-community/datetimepicker": "6.7.3",
-    "@react-native-community/netinfo": "9.3.7",
-    "@react-native-masked-view/masked-view": "0.2.8",
-    "@react-navigation/bottom-tabs": "5.11.15",
-    "@react-navigation/drawer": "5.12.9",
-    "@react-navigation/native": "5.9.8",
-    "@react-navigation/stack": "5.14.9",
-    "@sentry/react-native": "4.15.2",
+    "@react-native-async-storage/async-storage": "1.18.2",
+    "@react-native-community/datetimepicker": "7.2.0",
```
Comment (ardasnturk) [inline package.json:41]: This looks like a major update. Have you read the changelog for this package? How much does it affect our code?

File: package.json [hunk 2]
```diff
@@ -91,33 +91,33 @@
...
+    "react-native-safe-area-context": "4.6.3",
+    "react-native-screens": "~3.22.0",
     "react-native-snap-carousel": "4.0.0-beta.6",
     "react-native-status-bar-height": "^2.6.0",
-    "react-native-svg": "13.4.0",
-    "react-native-web": "~0.18.11",
-    "react-native-webview": "11.26.0",
+    "react-native-svg": "13.9.0",
+    "react-native-web": "~0.19.6",
+    "react-native-webview": "13.2.2",
```
Comment (ardasnturk) [inline package.json:118]: This looks like a major update. Have you read the changelog for this package? How much does it affect our code?

File: package.json [hunk 3]
```diff
@@ -91,33 +91,33 @@
...
+    "react-native-get-random-values": "~1.9.0",
     "react-native-gifted-chat": "^2.0.1",
-    "react-native-maps": "1.3.2",
+    "react-native-maps": "1.7.1",
     "react-native-markdown-display": "^7.0.0-alpha.2",
     "react-native-modal-dropdown": "https://github.com/donni106/react-native-modal-dropdown#feature/search-options",
     "react-native-progress": "^5.0.0",
     "react-native-qrcode-svg": "^6.1.2",
-    "react-native-reanimated": "~2.14.4",
+    "react-native-reanimated": "~3.3.0",
```
Comment (ardasnturk) [inline package.json:110]: This looks like a major update. Have you read the changelog for this package? How much does it affect our code?

File: package.json [hunk 4]
```diff
@@ -91,33 +91,33 @@
     "react": "18.2.0",
     "react-apollo": "~3.1.5",
     "react-dom": "18.2.0",
-    "react-hook-form": "^6.11.0",
-    "react-native": "0.71.8",
+    "react-hook-form": "^7.12.2",
```
Comment (ardasnturk) [inline package.json:94]: This looks like a major update. Have you read the changelog for this package? How much does it affect our code?

File: src/navigation/screenOptions.tsx [hunk 5]
```diff
@@ -67,9 +67,10 @@ const styles = StyleSheet.create({
     lineHeight: normalize(29)
   },
   headerTitleContainerStyle: {
-    flex: 1,
+    flex: 500,
```
Comment (ardasnturk) [inline src/navigation/screenOptions.tsx:70]: Why is the "flex: 500" style important to us? Why did you make this change?

**Review by ardasnturk — COMMENTED** [review]
I think the changelog of some packages should be checked.

File: package.json (hunk 1 above)
Comment (d4v3d4vE) [reply package.json:41]: It has no effects on the code, it just supports the new possible architecture of react native: https://reactnative.dev/docs/next/the-new-architecture/why

File: package.json (hunk 2 above)
Comment (d4v3d4vE) [reply package.json:118]: same here, no major changes

File: package.json (hunk 3 above)
Comment (d4v3d4vE) [reply package.json:110]: same here: "Reanimated 3.x doesn't introduce any breaking changes between 2.x and 3.x in terms of the API. All the code you've written in Reanimated v2 API works in 3.x without any changes."

File: package.json (hunk 4 above)
Comment (d4v3d4vE) [reply package.json:94]: also here, no changes effecting our code

File: src/navigation/screenOptions.tsx (hunk 5 above)
Comment (ardasnturk) [reply src/navigation/screenOptions.tsx:70]: And is there any particular reason why it's 500? Maybe 400? Or 0? Is it written in a document that we need to make the flex value 500?

**Comment by ardasnturk** [comment]
I've encountered some errors. I'm sharing screenshots below. Can you look at these errors more carefully?

|Android|Android & iOS|
|--|--|
the pictures in the app do not load on Android. Screenshot_20230822-141724 | An error occurs in the checkbox component as follows. The problem occurs on both platforms. Screenshot_20230822-141443

File: src/navigation/screenOptions.tsx (hunk 5 above)
Comment (d4v3d4vE) [reply src/navigation/screenOptions.tsx:70]: there was a slight difference between 400 and 500 and I figured out that it has no negative effects if the flex value is as big as this so I chose 500.
Flex 0 is also a option but it centers the header and is not positioned more on the left side like before:
| Flex 0 | Flex 500 |
| ------------- | ------------- |
| simulator_screenshot_A6CF4E25-7AA9-4429-A4F1-51BE8C62CFA2 |  |

Comment (donni106) [reply src/navigation/screenOptions.tsx:70]: what about no flex at all and removing `, alignItems: 'center'`?

Comment (d4v3d4vE) [reply src/navigation/screenOptions.tsx:70]: seems to be the same result as with `flex: 0`

Comment (d4v3d4vE) [reply src/navigation/screenOptions.tsx:70]: but the main issue is the spacing in the header with some buttons in some screens:
| Flex 0 / no Flex | Flex 500 |
| ------------- | ------------- |
| Simulator Screenshot - iPhone 14 Pro - 2023-08-23 at 10 56 56 | Simulator Screenshot - iPhone 14 Pro - 2023-08-23 at 10 57 19 |

**Comment by d4v3d4vE** [comment]
> 

both issues are fixed now, first two were because of react-hook-form [commit](https://github.com/smart-village-solutions/smart-village-app-app/pull/901/commits/ee4fe3d8cd15380f5ac58b16faa2cc4ae53107f9)
third one was because some SVG changes, adjusted the SVG-Icon now that it works and kept the same design [commit](https://github.com/smart-village-solutions/smart-village-app-app/pull/901/commits/b23472bf9eb5e1fffe68847dcd7b6f5fcfdf37bc)

| SE  | 14 Pro |
| ------------- | ------------- |
| Simulator Screenshot - iPhone SE (3rd generation) - 2023-08-25 at 15 11 59 | Simulator Screenshot - iPhone 14 Pro - 2023-08-25 at 15 12 52 |

**Comment by ardasnturk** [comment]
I just want to ask, do we have to update packages other than Expo on this branch? I think it would be better to create another branch to update other packages. If there is a package that needs to be updated with Expo 49, of course we should do it in this branch. At the moment the application for Android is crashing and I can't figure out where the problem is coming from. I don't know if it's the current Expo packages or other updated packages. Please update only the packages that need to be updated with Expo 49 in this branch. We can then create another branch for other package updates. This way I can test it more easily and better analyse where the problem is coming from.

Currently, the build I created for Android with the `npx expo run:android -d --variant Release` command crashes. Can you create a working android version for me with this command?

**Comment by d4v3d4vE** [comment]
> 

Thats weird, I couldn't build it with the command you've provided but I have done a working build with android studio

**Comment by ardasnturk** [comment]
> Thats weird, I couldn't build it with the command you've provided but I have done a working build with android studio

The build we create with Android studio creates a development build for us. With the command I sent, we create a production build. Therefore, we may encounter errors that we do not encounter in the development build in the production build. As a result, we will need a production build to publish the application in the store. So the priority for such major updates will always be to test with the production build. 👍

**Comment by d4v3d4vE** [comment]
> 

I've checked everything and this current branch already has the minimum updates required to ensure the app is running with Expo49

File: src/components/defectReport/DefectReportCreateForm.tsx [hunk 6]
```diff
@@ -257,10 +257,10 @@ export const DefectReportCreateForm = ({
 
         <Controller
           name="termsOfService"
-          render={({ onChange, value }) => (
+          render={({ field }) => (
             <Checkbox
-              checked={!!value}
-              onPress={() => onChange(!value)}
+              checked={!!field.value}
+              onPress={() => field.onChange(!field.value)}
```
Comment (ardasnturk) [inline src/components/defectReport/DefectReportCreateForm.tsx:263]: In some parts of the code `{ field: { name, onChange, value }` is used in some parts. Can you edit the whole code in the same format?

**Review by ardasnturk — CHANGES_REQUESTED** [review]

File: src/components/defectReport/DefectReportCreateForm.tsx (hunk 6 above)
Comment (d4v3d4vE) [reply src/components/defectReport/DefectReportCreateForm.tsx]: I've unified the code in this [commit](https://github.com/smart-village-solutions/smart-village-app-app/pull/901/commits/8753d75e2e49f895b382e945e241394c0a74dd4b)

**Review by ardasnturk — APPROVED** [review]

**Comment by donni106** [comment]
i rebased onto master and i am about to add some fine tuning.
among other things the test suite is not running. i will try to fix as much as possible and update some snapshots if succeeding.

File: src/components/defectReport/DefectReportCreateForm.tsx [hunk 7]
```diff
@@ -152,7 +152,7 @@ export const DefectReportCreateForm = ({
                 label: `${texts.defectReport.categoryName} *`,
                 placeholder: texts.defectReport.categoryName,
                 control
-              }}
+              }}
```
Comment (donni106) [inline src/components/defectReport/DefectReportCreateForm.tsx:155]: ? i will remove that

File: src/icons/editSetting.js [hunk 8]
```diff
@@ -1,7 +1,6 @@
...
-  <?xml version="1.0" encoding="UTF-8"?>
-  <svg width="16px" height="16px" viewBox="0 0 16 16" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
-    <path d="M8 10.5C9.38071 10.5 10.5 9.38071 10.5 8C10.5 6.61929 9.38071 5.5 8 5.5C6.61929 5.5 5.5 6.61929 5.5 8C5.5 9.38071 6.61929 10.5 8 10.5Z" stroke="${color}" stroke-miterlimit="10" stroke-linecap="round" stroke-linejoin="round"/>
-    <path d="M15.5 9V7L13.323 6.637C13.192 6.125 12.991 5.643 12.728 5.2L14.011 3.404L12.596 1.99L10.8 3.272C10.357 3.009 9.875 2.808 9.363 2.677L9 0.5H7L6.637 2.677C6.125 2.808 5.643 3.009 5.2 3.272L3.404 1.99L1.99 3.404L3.272 5.2C3.009 5.643 2.808 6.125 2.677 6.637L0.5 7V9L2.677 9.363C2.808 9.875 3.009 10.357 3.272 10.8L1.99 12.596L3.404 14.01L5.2 12.728C5.643 12.991 6.125 13.192 6.637 13.323L7 15.5H9L9.363 13.323C9.875 13.192 10.357 12.991 10.8 12.728L12.596 14.011L14.01 12.597L12.728 10.8C12.991 10.357 13.192 9.875 13.323 9.363L15.5 9Z" stroke="${color}" stroke-miterlimit="10" stroke-linecap="round" stroke-linejoin="round"/>
-  </svg>
-`;
+<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none">
```
Comment (donni106) [inline src/icons/editSetting.js:2]: i will re add `<?xml version="1.0" encoding="UTF-8"?>` as every icon file has that. and format the missing indentation

Comment (donni106) [reply src/icons/editSetting.js:2]: f455ac6

**Review by donni106 — APPROVED** [review]

File: src/navigation/MainTabNavigator.tsx [hunk 9]
```diff
@@ -14,14 +14,14 @@ export const MainTabNavigator = ({
...
-      inactiveBackgroundColor: tabNavigatorConfig.inactiveBackgroundColor,
-      tabStyle: { marginTop: normalize(0) },
-      style: { backgroundColor: colors.surface },
-      keyboardHidesTabBar: device.platform === 'android'
+    screenOptions={{
+      tabBarActiveTintColor: tabNavigatorConfig.activeTintColor,
+      tabBarInactiveTintColor: tabNavigatorConfig.inactiveTintColor,
+      tabBarActiveBackgroundColor: tabNavigatorConfig.activeBackgroundColor,
+      tabBarInactiveBackgroundColor: tabNavigatorConfig.inactiveBackgroundColor,
+      tabBarStyle: { backgroundColor: colors.surface, marginTop: normalize(0) },
```
Comment (donni106) [inline src/navigation/MainTabNavigator.tsx:22]: according to https://reactnavigation.org/docs/upgrading-from-5.x/#the-tabbaroptions-prop-is-removed-in-favor-of-more-flexible-options-for-bottom-tabs the `tabStyle` should have been migrated to `tabBarItemStyle`, right?

**Pull request merged**